tools/2023-11-interactive-commands-functional-tests.md

```diff
@@ -0,0 +1,124 @@
+# Title
+
+* **Status**: Pending
+* **Author**: Yetkin Timocin (@ytimocin)
+
+## Overview
+
+Radius CLI has multiple interactive commands that provide very important value to its users. Because these commands are very important for the functionality of Radius, we must make sure that we are able to cover different scenarios of these commands end to end. This design document will be about adding a tool to add functional tests for interactive Radius commands.
+
+## Terms and definitions
+
+Interactive Command: A command that needs to get user input. For example `rad init`.
+Pseudo-terminal: A pseudo-terminal is a software emulation of physical terminal that can be used to control terminal applications programmatically. We are going to be using pty and vt10x.
+
+## Objectives
+
+> **Issue Reference:** <https://github.com/radius-project/radius/issues/6572>
+
+### Goals
+
+* Add a tooling so that adding functional/e2e tests for interactive Radius commands is very easy for all the contributors.
+* Add a tooling so that adding functional/e2e tests for different paths of a given interactive Radius command is very easy for all the contributors.
+* Add functional/e2e tests for the interactive commands so that we know if something is broken.
```

Comment:
I appreciate that you called out the goals very clearly because now we can have a discussion about whether they are right goals.

I think the intro to this doc needs to be redone to explain:

- Our current strategy for testing `rad init`.
- What new information we learned (what are the problems with the current strategy?)
- Why we need an investment in new tools and techniques.
- What changes should we make to our existing strategy.

We keep a high bar for changing a previous decision. We also keep a high bar for adding new tools and techniques.